.github/workflows/publish-page.yml

```diff
@@ -19,7 +19,7 @@ jobs:
         documentation_path: './docs/source'
     - name: Preserve Configuration Files
       run: |
-        git checkout main -- CODEOWNERS
+        echo "* @mike-petersen-ni" > CODEOWNERS
```

Comment:
This is hacky, but basically when the gh-pages build occurs, it blows away the old files and replaces all of them. This ensures that the files which aren't outputs of that build process are preserved - specifically the GitHub configuration files needed for code owners and CNAME redirection.

There's probably a better way to do this, such as making those files an official build output.

Reply:
Huh, looks like this new option is in Beta: https://github.com/actions/deploy-pages 
Maybe worth a look?

Reply:
If this PR is changing `main`, why wouldn't the `git checkout main` work? (In other words, why replace this with `echo`?)

Reply:
Once we add Kayla to the `CODEOWNERS` file, it would end up adding her on the html publishing PRs too. I don't think we want her to be an owner on those PRs. Using `echo` ensures the `CODEOWNERS` in the `gh-pages` branch (which is what is used for PRs to that branch) only has you. 

Honestly, there's probably a better way to handle this. I'm not convinced having those publishing PRs reviewed is really necessary anymore now that things have been running for a while but it might take some finagling to get them to autopublish.